### PR TITLE
Remove `assert_nothing_raised`.

### PR DESCRIPTION
**Pull request by tgxworld**
See discussion in https://github.com/seattlerb/minitest/issues/159.

cc/ @tenderlove

### CODE REVIEW

**Comment by rafaelfranca** [comment]
We are keeping this method because it is intention revealing. With it you know that you are calling an method expecting it to not raise. Without it we don't know.

I agree we should avoid using it when we can, but sometimes it is handy. For that reason I prefer to keep it.

File: activesupport/lib/active_support/test_case.rb [hunk 1]
```diff
@@ -72,14 +72,5 @@ def test_order
     alias :assert_not_predicate :refute_predicate
     alias :assert_not_respond_to :refute_respond_to
     alias :assert_not_same :refute_same
-
-    # Fails if the block raises an exception.
-    #
-    #   assert_nothing_raised do
-    #     ...
-    #   end
-    def assert_nothing_raised(*args)
```
Comment (tgxworld) [inline activesupport/lib/active_support/test_case.rb]: Some history behind this method. http://blog.zenspider.com/blog/2012/01/assert_nothing_tested.html

**Comment by rafaelfranca** [comment]
Forget to say. Thank you for the pull request. :heart:

**Comment by tgxworld** [comment]
> We are keeping this method because it is intention revealing. With it you know that you are calling an method expecting it to not raise. Without it we don't know.

I do agree that it is intention revealing but it seems odd that we're advocating for a method that serves no purpose when ["you always expect your code to not raise an exception." ](https://github.com/seattlerb/minitest/issues/159#issuecomment-8214938).

**Comment by tgxworld** [comment]
For instance 
```ruby
   test "method call to mail does not raise error" do
     assert_nothing_raised { BaseMailer.welcome }
   end
```
Using `assert_nothing_raised` here serves no purpose at all and isn't that much more intentional revealing than just running the test and not seeing any exceptions.

**Comment by rafaelfranca** [comment]
But sometimes you want to make explicit the system under test, in the case where `assert_nothing_raised` is used just the call that you expect to not raise an exception.

The code that you posted without the assertion would look like someone forget to add the exception and I'm almost sure people would send PRs to remove the tests if it is just:

```
test "method call to mail does not raise error" do
  BaseMailer.welcome
end
```

**Comment by rafaelfranca** [comment]
BTW, I'm one of the people that ask people to avoid `assert_nothing_raised` in most of the PRs, but sometimes it is needed and I prefer to keep it.

**Comment by tgxworld** [comment]
@rafaelfranca Ok! :smile:

**Comment by tgxworld** [comment]
@rafaelfranca On a similar note, should we be recommending the method in our docs?
https://github.com/rails/rails/blob/master/activejob/lib/active_job/test_helper.rb#L154

**Pull request closed without merging**